build/source/engine/allocspace.f90

```diff
@@ -21,32 +21,31 @@
 module allocspace_module
 ! data types
 USE nrtype
-! metadata structures
-USE globalData,only:time_meta,forc_meta,attr_meta,type_meta        ! metadata structures
-USE globalData,only:prog_meta,diag_meta,flux_meta,deriv_meta       ! metadata structures
-USE globalData,only:mpar_meta,indx_meta                            ! metadata structures
-USE globalData,only:bpar_meta,bvar_meta                            ! metadata structures
-USE globalData,only:model_decisions                                ! model decision structure
-! time structures
-USE globalData,only:refTime,startTime,finshTime                    ! reference time, start time, and end time for the model simulation
-! structure information
-USE globalData,only:structInfo             ! information on the data structures                  
 ! provide access to the derived types to define the data structures
 USE data_types,only:&
+                    ! no spatial dimension
                     var_i,               & ! x%var(:)            (i4b)
                     var_d,               & ! x%var(:)            (dp)
                     var_ilength,         & ! x%var(:)%dat        (i4b)
                     var_dlength,         & ! x%var(:)%dat        (dp)
-                    spatial_int,         & ! x%hru(:)%var(:)     (i4b)
-                    spatial_double,      & ! x%hru(:)%var(:)     (dp)
-                    spatial_intVec,      & ! x%hru(:)%var(:)%dat (i4b)
-                    spatial_doubleVec      ! x%hru(:)%var(:)%dat (dp)
+                    ! gru dimension
+                    gru_int,             & ! x%gru(:)%var(:)     (i4b)
+                    gru_double,          & ! x%gru(:)%var(:)     (dp)
+                    gru_intVec,          & ! x%gru(:)%var(:)%dat (i4b)
+                    gru_doubleVec,       & ! x%gru(:)%var(:)%dat (dp)
+                    ! gru+hru dimension
+                    gru_hru_int,         & ! x%gru(:)%hru(:)%var(:)     (i4b)
+                    gru_hru_double,      & ! x%gru(:)%hru(:)%var(:)     (dp)
+                    gru_hru_intVec,      & ! x%gru(:)%hru(:)%var(:)%dat (i4b)
+                    gru_hru_doubleVec      ! x%gru(:)%hru(:)%var(:)%dat (dp)
 ! metadata structure
-USE data_types,only:var_info
+USE data_types,only:var_info               ! data type for metadata
+USE globalData,only:gru_struc              ! gru-hru mapping structures
 implicit none
 private
-public::initStruct
+public::allocGlobal
 public::allocLocal
+public::allocate_gru_struc
 ! define missing values
 integer(i4b),parameter :: missingInteger=-9999
```

Comment:
This is defined as a global, which seems necessary here, since the purpose is a program-wide flag.